cranelift-codegen/meta/src/isa/x86/recipes.rs

```diff
@@ -3035,10 +3036,87 @@ pub(crate) fn define<'shared>(
             r#"
-            sink.add_stackmap(args, func, isa);
-        "#,
+                sink.add_stackmap(args, func, isa);
+            "#,
         ),
     );
 
+    /*recipes.add_recipe(
+        EncodingRecipeBuilder::new("tls_value", &formats.unary_global_value, 19)
+            .operands_out(vec![gpr])
+            .emit(
+                r#"
+                    // output %rax
+                    // clobbers %rdi
+
+                    // leaq @tlsld(%rip), %rdi
+                    sink.put1(0x48);
+                    sink.put1(0x8d);
+                    sink.put1(0x3d);
+                    sink.reloc_external(Reloc::ElfX86_64TlsLd,
+                                        &func.global_values[global_value].symbol_name(),
+                                        -4);
+                    sink.put4(0);
+
+                    // callq __tls_get_addr-4
+                    sink.put1(0xe8);
+                    sink.reloc_external(Reloc::X86CallPLTRel4,
+                                        &ExternalName::LibCall(LibCall::ElfTlsGetAddr),
+                                        -4);
+                    sink.put4(0);
+
+                    // leaq @dtpoff32(%rax), %rax
+                    sink.put1(0x48);
+                    sink.put1(0x8d);
+                    sink.put1(0x80);
+                    sink.reloc_external(Reloc::ElfX86_64DtpOff32,
+                                        &func.global_values[global_value].symbol_name(),
+                                        0);
+                    sink.put4(0);
+                "#,
+            ),
+    );*/
+
+    recipes.add_recipe(
+        EncodingRecipeBuilder::new("elf_tlsld", &formats.unary_global_value, 7)
+            // FIXME Should use gpr instead of reg_rdi, but that causes an unnecessary regmove instruction.
+            .operands_out(vec![reg_rdi])
+            .emit(
+                r#"
+                    // leaq @tlsld(%rip), %rdi
+                    sink.put1(0b0100_1000);
+                    const LEA: u8 = 0x8d;
+                    sink.put1(LEA);
+                    modrm_riprel(0b111 /*out_reg0*/, sink);
+                    sink.reloc_external(Reloc::ElfX86_64TlsLd,
+                                        &func.global_values[global_value].symbol_name(),
+                                        -4);
+                    sink.put4(0);
+                "#,
+            ),
+    );
+
+    recipes.add_recipe(
+        EncodingRecipeBuilder::new("elf_dtpoff32", &formats.tls_offset, 7)
+            // FIXME Correct encoding for non rax registers
+            .operands_in(vec![reg_rax])
+            .operands_out(vec![reg_rax])
+            .emit(
+                r#"
+                    // output %rax
+
+                    // leaq @dtpoff32(%rax), %rax
+                    sink.put1(0b0100_1000);
+                    const LEA: u8 = 0x8d;
+                    sink.put1(LEA);
+                    modrm_disp32(0b000/*in_reg0*/, 0b000/*out_reg0*/, sink);
```

Comment:
Were you planning to actually use `in_reg0` and `out_reg0` here but temporarily hard-coded the `RegUnit`s?